docker-compose.yml

```diff
@@ -59,7 +59,7 @@ services:
       - redis
 
   search:
-    image: elasticsearch:5.4.3
+    image: elasticsearch:6.4.2
```

Comment:
This image seems to originate from `docker.elastic.co/elasticsearch/elasticsearch:6.4.1` as per `https://github.com/docker-library/elasticsearch/blob/cf9cba0a1590cfd894d51bc7998d9d36b2d47afd/6.4.1/Dockerfile`

Usage of the oss flavor is preferable.  `docker.elastic.co/elasticsearch/elasticsearch-oss:6.4.2`
X-pack etc should be not included by default imo.

Read more [here]( https://www.elastic.co/products/x-pack/open)

Reply:
I'm not sure I understand why having fewer tools is better other than saving a couple megabytes in file size.

Reply:
The issue is that x-pack is not free and without a mechanism in the Dockerfile to disable it, some users would be unaware.

```
If the code of X-Pack is open, does that mean it's all free?

No ... Some features in X-Pack are paid, and require a license that comes with a Gold or Platinum subscription.
```

Reply:
It may not give you all features for free but it's a genuinely useful tool. What problems are there related to using X-Pack under the Elastic License?

Reply:
@stephenmoloney We discussed it internally - we're fine with that license and we prefer to use the official Elasticserach image.

Reply:
yea -sure. :+1: for updating elasticsearch.

If using the stock image, I'd suggest adding the xpack env configuration to docker-compose:

```
search:
    image: elasticsearch:6.4.2
    restart: unless-stopped
    networks:
      - saleor-backend-tier
    volumes:
      - saleor-search:/usr/share/elasticsearch/
    ports:
      - 9200:9200
    # See https://github.com/docker/compose/issues/4513 if updating to version '3'
    mem_limit: 512m
    environment:
      - "ES_JAVA_OPTS=-Xms512m -Xmx512m -Dlog4j2.disable.jmx=true"
     - "xpack.license.self_generated.type=basic"
     - "xpack.ml.enabled=false"
     - "xpack.monitoring.enabled=false"
     - "xpack.security.enabled=false"
    - "xpack.security.audit.enabled=false"
     - "xpack.watcher.enabled=false"
```

I haven't tested this.

Reply:
I was following the [Elasticsearch docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html) when upgrading.

Reply:
Here are configuration options pages I found for reference if it's useful to anyone

| feature  | link  |
|---|---|
| machine learning  |  https://www.elastic.co/guide/en/elasticsearch/reference/6.4/ml-settings.htm l
| monitoring | https://www.elastic.co/guide/en/elasticsearch/reference/current/configuring-monitoring.html  |
| security  | https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html  |
| security auditing   | https://www.elastic.co/guide/en/elasticsearch/reference/current/auditing-settings.html  |
| watcher  | https://www.elastic.co/guide/en/elasticsearch/reference/6.4/notification-settings.html  |
| license type  | https://www.elastic.co/guide/en/elasticsearch/reference/6.4/license-settings.html  |

but i think it would be nice to have it in the `docker-compose` simply for visibility.